homeassistant/components/light/__init__.py

```diff
@@ -292,17 +293,68 @@ class SetIntentHandler(intent.IntentHandler):
                 pars[ATTR_PROFILE] = Profiles.get_default(light.entity_id)
                 preprocess_turn_on_alternatives(pars)
                 turn_light_off, off_pars = preprocess_turn_off(pars)
+
             if turn_light_off:
-                await light.async_turn_off(**off_pars)
+                light_calls.append(light.async_turn_off(**off_pars))
             else:
-                await light.async_turn_on(**pars)
+                light_calls.append(light.async_turn_on(**pars))
+
+            if not light.should_poll:
+                continue
+
+            update_tasks.append(
+                light.async_update_ha_state(True))
+
+        if light_calls:
+            await asyncio.gather(*light_calls, loop=hass.loop)
+
+        if update_tasks:
+            await asyncio.wait(update_tasks, loop=hass.loop)
+
+    async def async_handle_light_off_service(service):
+        """Handle a turn light off service call."""
+        # Get the validated data
+        params = service.data.copy()
+
+        # Convert the entity ids to valid light ids
+        target_lights = await component.async_extract_from_service(service)
+        params.pop(ATTR_ENTITY_ID, None)
+
+        if service.context.user_id:
+            user = await hass.auth.async_get_user(service.context.user_id)
+            if user is None:
+                raise UnknownUser(context=service.context)
+
+            entity_perms = user.permissions.check_entity
+
+            for light in target_lights:
+                if not entity_perms(light, POLICY_CONTROL):
+                    raise Unauthorized(
+                        context=service.context,
+                        entity_id=light,
+                        permission=POLICY_CONTROL
+                    )
+        light_calls = []
+        update_tasks = []
+        for light in target_lights:
+            light.async_set_context(service.context)
+
+            pars = params
+            if not pars:
+                pars = params.copy()
+                pars[ATTR_PROFILE] = Profiles.get_default(light.entity_id)
```

Comment:
Should we add a profile for _turn off_ calls? I mean i think it only makes sense for turn on I think.

Reply:
Cheers,  the linting appears to be passing now, however there seems to be a single test case failing for the rflink component:

Reply:
I won't be able to debug it until next week but I'm guessing it's something to do with the repetitions and the calls now being executed concurrently.  I've not used the rflink component before so I'll need to do some digging.

If my hunch is correct and the rflink test case relies on multiple calls being run non-concurrently then I'm not sure how this could be resolved.  If anyone has any thoughts please let me know.